src/Caching/Hybrid/src/Runtime/HybridCache.cs

```diff
@@ -0,0 +1,119 @@
+
+    /// <summary>
+    /// Get data from the cache, or the underlying data service if not available.
+    /// </summary>
+    /// <typeparam name="T">The type of the data being considered</typeparam>
+    /// <param name="key">The unique key for this cache entry</param>
+    /// <param name="underlyingDataCallback">Provides the underlying data service is the data is not available in the cache</param>
+    /// <param name="options">Additional options for this cache entry</param>
+    /// <param name="tags">The tags to associate with this cache item</param>
+    /// <param name="token">Cancellation for this operation</param>
+    /// <returns>The data, either from cache or the underlying data service</returns>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("ApiDesign", "RS0026:Do not add multiple public overloads with optional parameters", Justification = "Delegate differences make this unambiguous")]
+    public ValueTask<T> GetOrCreateAsync<T>(string key, Func<CancellationToken, ValueTask<T>> underlyingDataCallback,
+        HybridCacheEntryOptions? options = null, ICollection<string>? tags = null, CancellationToken token = default)
+        => GetOrCreateAsync(key, underlyingDataCallback, WrappedCallbackCache<T>.Instance, options, tags, token);
+
+    private static class WrappedCallbackCache<T> // per-T memoized helper that allows GetOrCreateAsync<T> and GetOrCreateAsync<TState, T> to share an implementation
+    {
+        // for the simple usage scenario (no TState), pack the original callback as the "state", and use a wrapper function that just unrolls and invokes from the state
+        public static readonly Func<Func<CancellationToken, ValueTask<T>>, CancellationToken, ValueTask<T>> Instance = static (callback, ct) => callback(ct);
+    }
+
+    /// <summary>
+    /// Manually insert or overwrite a cache entry.
+    /// </summary>
+    /// <typeparam name="T">The type of the data being considered</typeparam>
+    /// <param name="key">The unique key for this cache entry</param>
+    /// <param name="value">The value to assign for this cache item</param>
+    /// <param name="options">Additional options for this cache entry</param>
+    /// <param name="tags">The tags to associate with this cache item</param>
+    /// <param name="token">Cancellation for this operation</param>
+    public abstract ValueTask SetAsync<T>(string key, T value, HybridCacheEntryOptions? options = null, ICollection<string>? tags = null, CancellationToken token = default);
+
+    /// <summary>
+    /// Removes cache data with the specified key
+    /// </summary>
+    public abstract ValueTask RemoveKeyAsync(string key, CancellationToken token = default);
+
+    /// <summary>
+    /// Removes cache data with the specified keys
+    /// </summary>
+    public virtual ValueTask RemoveKeysAsync(IEnumerable<string> keys, CancellationToken token = default)
```

Comment:
Personally, I like to make all members either abstract or sealed because I find it makes the code easier to reason about.  It's less important for members than for types though.  Do you have a potential override in mind?

Reply:
yes, if a different implementation supports multi-key remove cleanly

Reply:
Works for me.